packages/lib/src/__tests__/input.test.ts

```diff
@@ -0,0 +1,97 @@
+import { NoLogLogger } from '@smartthings/core-sdk/dist/logger'
+
+import { FileInputProcessor, formatFromFilename, IOFormat, parseJSONOrYAML } from '../input'
+import { LogManager } from '../logger'
+
+
+const resourcesDir = './src/__tests__/resources'
+
+interface SimpleType {
+	num: number
+	str: string
+}
+
+const validYAML = 'num: 5\nstr: my string'
+const validData = { num: 5, str: 'my string' }
+
+describe('formatFromFilename', () => {
+	it('handles yaml extensions', function() {
+		expect(formatFromFilename('fn.yaml')).toBe(IOFormat.YAML)
+		expect(formatFromFilename('fn.yml')).toBe(IOFormat.YAML)
+		expect(formatFromFilename('fn.YAML')).toBe(IOFormat.YAML)
+	})
+
+	it('handles json extensions', function() {
+		expect(formatFromFilename('fn.json')).toBe(IOFormat.JSON)
+		expect(formatFromFilename('fn.JSON')).toBe(IOFormat.JSON)
+	})
+
+	it('defaults to YAML', function() {
+		jest.spyOn(LogManager.prototype, 'getLogger').mockImplementation(() => {
+			return new NoLogLogger()
+		})
+		expect(formatFromFilename('fn')).toBe(IOFormat.YAML)
+	})
+})
+
+describe('parseJSONOrYAML', () => {
+	it('returns data from valid yaml', function() {
+		const result = parseJSONOrYAML(validYAML, 'source')
+
+		expect(result).toEqual(validData)
+	})
+
+	it('throws error with no data', function() {
+		expect(() => parseJSONOrYAML('', 'empty')).toThrow('did not get any data from empty')
+	})
+
+	it('throws error with simple string data', function() {
+		expect(() => parseJSONOrYAML('just a string', 'empty')).toThrow('got simple string from empty')
+	})
+})
+
+// TODO: readDataFromStdin
+/*
+describe('readDataFromStdin', () => {
+	it('returns data from valid yaml', async function() {
+		const buf = Buffer.from(validYAML, 'utf-8')
+		jest.spyOn(process.stdin, 'on')
+			.mockRejectedValueOnce(buf)
+		expect(readDataFromStdin()).resolves.toBe(validYAML)
+	})
+
+	it('passes on error', function() {
+		expect(() => parseJSONOrYAML('', 'empty')).toThrow('did not get any data from empty')
+	})
+})
+*/
```

Comment:
I considered using that but it's so old I was a little wary of it. I actually did get this working already without it but I'll add playing with that to my list because it might be better than what I've done (which involves some `@ts-ignore`s.

Reply:
Got this to work and code has been updated.